src/Pos/Delegation/Types.hs

```diff
@@ -68,3 +69,9 @@ type DlgMemPool = HashMap PublicKey ProxySKHeavy
 
 -- | Confirmation of light cert type.
 type ProxySKLightConfirmation = (ProxySKLight, ProxySigLight ProxySKLight)
+
+-- | Lightweight PSK or heavyweight PSK with real leader public key
+-- (because heavyweight psks have redelegation feature, so pskIssuerPk
+-- hPsk /= leader in general case). This is used to create a block
+-- header only.
+type ProxySKBlockInfo = Either ProxySKLight (ProxySKHeavy, PublicKey)
```

Comment:
1. I don't really want to turn it into the datatype, because it'll make things more complicated IMO. More constructors, similar to `BlockSignature`. Eh. 
2. I thought about wrapping it into the newtype, but i didn't get the point. It's anyway wrapped only once and unwrapped once as well.
3. Wrapping it into `Maybe` makes sense to me.